core/src/main/scala/dagr/core/execsystem/TaskExecutionRunner.scala

```diff
@@ -47,7 +47,7 @@ private object TaskRunner {
     * @param script the location of the task's script
     * @param logFile the location of the task's log file
     */
-  class SingleTaskRunner(task: ProcessTask, script: Path, logFile: Path) extends TaskRunnable(task = task) {
+  class SingleTaskExecutionRunner(task: ProcessTask, script: Path, logFile: Path) extends TaskRunnable(task = task) {
```

Comment:
It won't let me comment on the appropriate line since you didn't change it, but I think it would be clearer if on interruption of the thread you set the return value to some less commonly used value than 1 (e.g. 255) and also had a constant for that so it's clear elsewhere.